src/inference/graph_inference.cpp

```diff
@@ -361,23 +394,38 @@ class GraphNodeAssignment : public Nice2Assignment {
   double GetNodeScore(const GraphInference& fweights, int node) const {
     double sum = -GetNodePenalty(node);
     const GraphInference::FeaturesMap& features = fweights.features_;
+    const GraphInference::FactorFeaturesMap& factor_features = fweights.factor_features_;
     for (const GraphQuery::Arc& arc : query_->arcs_adjacent_to_node_[node]) {
-       GraphFeature feature(
-            assignments_[arc.node_a].label,
-            assignments_[arc.node_b].label,
-            arc.type);
-       auto feature_it = features.find(feature);
-       if (feature_it != features.end()) {
-         sum += feature_it->second.getValue();
-       }
-     }
-     return sum;
+      GraphFeature feature(
+          assignments_[arc.node_a].label,
+          assignments_[arc.node_b].label,
+          arc.type);
+      auto feature_it = features.find(feature);
+      if (feature_it != features.end()) {
+        sum += feature_it->second.getValue();
+      }
+    }
+
+    for (uint i = 0; i < query_->factors_of_a_node_[node].size(); i++) {
```

Comment:
I put it because the compiler was giving me warnings of compare between unsigned and signed int. I think size_t is int so I will have the same warning I believe